File: code/python/command/PreparePizzaCommand.py
from Command import Command
from Receiver import Receiver

class PreparePizzaCommand(Command):
    def __init__(self, receiver):
        self.receiver = receiver

    def execute(self):
        print("PreparePizzaCommand.execute(): Delegating to Receiver.prepare_pizza()")
        self.receiver.prepare_pizza()

File: code/typescript/src/command/PreparePizzaCommand.ts
import { Command } from "./Command";
import { Receiver } from "./Receiver";

export class PreparePizzaCommand implements Command {
    private receiver: Receiver;

    constructor(receiver: Receiver) {
        this.receiver = receiver;
    }

    execute(): void {
        console.log("PreparePizzaCommand.execute(): Delegating to Receiver.preparePizza()");
        this.receiver.preparePizza();
    }
}
